cl/beacon/validatorapi/post.go

```diff
@@ -46,14 +54,105 @@ func (v *ValidatorApiHandler) PostEthV1ValidatorSyncCommitteeSubscriptions(w htt
 	return nil, beaconhttp.NewEndpointError(404, errNotImplemented)
 }
 
-func (v *ValidatorApiHandler) PostEthV1ValidatorBeaconCommitteeSubscriptions(w http.ResponseWriter, r *http.Request) (*int, error) {
+func (v *ValidatorApiHandler) PostEthV1ValidatorBeaconCommitteeSubscriptions(w http.ResponseWriter, r *http.Request) (*string, error) {
 	var req []building.BeaconCommitteeSubscription
 	err := json.NewDecoder(r.Body).Decode(&req)
 	if err != nil {
 		return nil, beaconhttp.NewEndpointError(400, fmt.Errorf("invalid request: %w", err))
 	}
-	// TODO: this endpoint
-	return nil, beaconhttp.NewEndpointError(404, errNotImplemented)
+
+	// the empty list i believe is a valid request, and it will just do nothing
+	if len(req) == 0 {
+		return nil, nil
+	}
+
+	l := log.New()
+
+	/*
+		After beacon node receives this request,
+		search using discv5 for peers related to this subnet and replace current peers with those ones if necessary.
+		If validator is_aggregator, beacon node must:
+
+		 1. announce subnet topic subscription on gossipsub
+		 2. aggregate attestations received on that subnet
+	*/
+
+	// grab the head state to calculate committee counts with
+	s, err := v.privateGetStateFromStateId("head")
```

Comment:
this is wrong. it should be sent to ProcessAttestation instead in the fcu.

Reply:
kk